lib/mayaUsd/ufe/UsdContextOps.cpp

```diff
@@ -90,6 +95,33 @@ class SetVariantSelectionUndoableCommand : public Ufe::UndoableCommand
     const std::string fNewSelection;
 };
 
+//! \brief Undoable command for prim active state change
+class ToggleActiveStateCommand : public Ufe::UndoableCommand
+{
+public:
+    ToggleActiveStateCommand(const UsdPrim& prim)
+        : _prim(prim)
+    {
+        _active = _prim.IsActive();
+    }
+
+    void undo() override
+    {
+        MayaUsd::ufe::InAddOrDeleteOperation ad;
+        _prim.SetActive(_active);
+    }
+
+    void redo() override
+    {
+        MayaUsd::ufe::InAddOrDeleteOperation ad;
+        _prim.SetActive(!_active);
+    }
+
+private:
+    UsdPrim _prim;
```

Comment:
On working with Hamed on parenting bugs I've become increasingly weary about storing prims (or scene items, same thing) as command data members, because they can go stale.  Imagine the following sequence of commands:
- Inactivate prim (prim stored in this command)
- Reparent prim (prim removed using UsdStage::removePrim())
- Undo reparent
- Undo inactivate --- boom.
You should be able to demonstrate this fairly easily.  Solution is the same everywhere, and I know you've implemented it in the past: use a path.  In this case it can be the SdfPath and the UsdStage of the prim (that would get my vote), or you can use the Ufe::Path and call Hierarchy::createItem(), then downcast to the UsdSceneItem and get the prim.

Reply:
Ah yes good thinking.

Reply:
See commit e19fa54bb6955fb94e66aef00a39385e250dd73a